src/main/java/com/example/bookshop/model/ShoppingCart.java

```diff
@@ -0,0 +1,32 @@
+package com.example.bookshop.model;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotNull;
+import java.util.Set;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Table(name = "shopping_cart")
+@SQLDelete(sql = "UPDATE shopping_cart SET  is_deleted = true WHERE id = ?")
+@SQLRestriction("status <> 'DELETED'")
+@Getter
+@Setter
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull
+    @OneToOne
```

Comment:
you can use @MapsId here 
```suggestion
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id", nullable = false)
```